### PR TITLE
Refactored XBHelpMenus and XBUtilities to remove xrt_core device dependencies

### PR DESCRIPTION
**Pull request by vboggara-xilinx**
#### Problem solved by the commit
Refactored XBHelpMenus and XBUtilities to remove xrt_core device dependencies.
*Divided  XBHelpMenus and XBUtilities files to 2 separate files, one is with XRT core dependencies and another one is without XRT core dependencies. 
*These changes required for XBFLASH standalone utility to utilize common code base.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
n/a
#### How problem was solved, alternative solutions (if any) and why they were rejected
n/a
#### Risks (if any) associated the changes in the commit
none
#### What has been tested and how, request additional testing if necessary
verified xbutil and xbmgmt with these changes locally.
#### Documentation impact (if any)
n/a

### CODE REVIEW

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good. @rozumx can suggest if there are any better names for these common files.

**Comment by vboggara-xilinx** [comment]
@rozumx , As discussed in today's meeting. included newly created (*Core.h) files to the files wherever needed. please review the changes and merge these changes.  This is blocking PR for XBFLASH standalone tool changes.

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by vboggara-xilinx** [comment]
@rozumx As discussed yesterday, This is blocking PR for XBFLASH standalone tool changes. please review and merge this PR at the earliest.

File: src/runtime_src/core/tools/common/OptionOptions.cpp [hunk 1]
```diff
@@ -20,8 +20,8 @@
 #include <iostream>
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/OptionOptions.cpp:20]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/Process.cpp [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #include "ProgressBar.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/Process.cpp:37]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/ReportHost.cpp [hunk 3]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportHost.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/XBMain.h [hunk 4]
```diff
@@ -20,6 +20,7 @@
 // Include files
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/XBMain.h:20]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/XBUtilities.h [hunk 5]
```diff
@@ -32,83 +32,22 @@
 #include <boost/algorithm/string.hpp>
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/XBUtilities.h:32]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/XBUtilitiesCore.h [hunk 6]
```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2019-2021 Xilinx, Inc
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/XBUtilitiesCore.h:2]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp [hunk 7]
```diff
@@ -16,6 +16,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp:16]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp [hunk 8]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdAdvanced.cpp [hunk 9]
```diff
@@ -24,6 +24,7 @@
 #include "common/device.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/SubCmdAdvanced.cpp:24]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp [hunk 10]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdReset.cpp [hunk 11]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/SubCmdReset.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp [hunk 12]
```diff
@@ -30,6 +30,7 @@
 #include "core/common/device.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:30]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp [hunk 13]
```diff
@@ -24,6 +24,7 @@
 #include "core/common/query_requests.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp:24]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbmgmt2/flash/xspi.cpp [hunk 14]
```diff
@@ -32,6 +32,7 @@
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbmgmt2/flash/xspi.cpp:32]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/CMakeLists.txt [hunk 15]
```diff
@@ -5,9 +5,11 @@
 file(GLOB XBUTIL_V2_BASE_FILES
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/CMakeLists.txt:5]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/OO_AieRegRead.cpp [hunk 16]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_AieRegRead.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp [hunk 17]
```diff
@@ -16,6 +16,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp:16]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/OO_MemRead.cpp [hunk 18]
```diff
@@ -18,6 +18,7 @@
 #include "OO_MemRead.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_MemRead.cpp:18]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp [hunk 19]
```diff
@@ -18,6 +18,7 @@
 #include "OO_MemWrite.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:18]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/OO_P2P.cpp [hunk 20]
```diff
@@ -16,6 +16,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_P2P.cpp:16]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/SubCmdProgram.cpp [hunk 21]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdProgram.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp [hunk 22]
```diff
@@ -19,6 +19,7 @@
 #include "xrt.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp:19]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/xbutil2/XBReport.cpp [hunk 23]
```diff
@@ -17,6 +17,7 @@
 // ------ I N C L U D E   F I L E S -------------------------------------------
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/XBReport.cpp:17]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/core/tools/common/Process.cpp [hunk 24]
```diff
@@ -37,6 +37,7 @@
 #include "ProgressBar.h"
 #include "Process.h"
 #include "EscapeCodes.h"
+#include "XBUtilitiesCore.h"
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/Process.cpp:40]: I like it that you initially added XBUtilitiesCore.h prior to XBUtilities.h.  Doing so helps greatly in verifying that there are no longer any dependencies between these two files.

File: src/runtime_src/core/tools/common/XBUtilities.cpp [hunk 25]
```diff
@@ -56,269 +57,21 @@ static const uint32_t FDT_END = 0x9;
...
-  uint32_t totalsize;
-  uint32_t off_dt_struct;
-  uint32_t off_dt_strings;
-  uint32_t off_mem_rsvmap;
-  uint32_t version;
-  uint32_t last_comp_version;
-  uint32_t boot_cpuid_phys;
-  uint32_t size_dt_strings;
-  uint32_t size_dt_struct;
+    uint32_t magic;
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/XBUtilities.cpp:60]: Why the extra spaces.  Currently, the code here is using 2 spaces for idention.

**Review by rozumx — APPROVED** [review]
Several comments (e.g., copyright date) and some format changes requested.  That said, no functionality or code changes needed.

Approved.

**Review by AShivangi — APPROVED** [review]

File: src/runtime_src/core/tools/common/OptionOptions.cpp (hunk 1 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/OptionOptions.cpp:20]: comment addressed.

File: src/runtime_src/core/tools/common/Process.cpp (hunk 2 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/Process.cpp:37]: comment addressed.

File: src/runtime_src/core/tools/common/Process.cpp (hunk 24 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/Process.cpp:40]: agreed.

File: src/runtime_src/core/tools/common/ReportHost.cpp (hunk 3 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/ReportHost.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/common/XBMain.h (hunk 4 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/XBMain.h:20]: comment addressed.

File: src/runtime_src/core/tools/common/XBUtilities.h (hunk 5 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/XBUtilities.h:32]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp (hunk 7 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp:16]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp (hunk 8 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdAdvanced.cpp (hunk 9 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/SubCmdAdvanced.cpp:24]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp (hunk 10 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/SubCmdReset.cpp (hunk 11 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/SubCmdReset.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp (hunk 12 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/flash/xmc.cpp:30]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp (hunk 13 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp:24]: comment addressed.

File: src/runtime_src/core/tools/xbmgmt2/flash/xspi.cpp (hunk 14 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbmgmt2/flash/xspi.cpp:32]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/OO_AieRegRead.cpp (hunk 16 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_AieRegRead.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp (hunk 17 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp:16]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/OO_MemRead.cpp (hunk 18 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_MemRead.cpp:18]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp (hunk 19 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:18]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/OO_P2P.cpp (hunk 20 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/OO_P2P.cpp:16]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/SubCmdProgram.cpp (hunk 21 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/SubCmdProgram.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp (hunk 22 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/SubCmdReset.cpp:19]: comment addressed.

File: src/runtime_src/core/tools/xbutil2/XBReport.cpp (hunk 23 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/xbutil2/XBReport.cpp:17]: comment addressed.

File: src/runtime_src/core/tools/common/XBUtilities.cpp (hunk 25 above)
Comment (vboggara-xilinx) [reply src/runtime_src/core/tools/common/XBUtilities.cpp]: Removed extra spaces.

**Comment by vboggara-xilinx** [comment]
@rozumx All comments addressed and build passed, please merge this change.

**Pull request merged**